File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    const scoreMapping = {}
//  Mapping해주기
    for (i=0; i < name.length; i++){
        scoreMapping[name[i]] = yearning[i]
    }
    console.log(scoreMapping)
// 추억 카운팅
    photo.forEach((nameList)=>{
        let count = 0
        nameList.forEach((name)=>{
            if(scoreMapping[name]){
                count = count + scoreMapping[name]
            }
        })
        answer.push(count)
    })
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        Map<String, Integer> missScore = new HashMap();
        // HashMap에 점수 저장
        for (int i=0; i<name.length; i++) missScore.put(name[i], yearning[i]);
        
        for (int i=0; i<photo.length; i++){
            for (String n: photo[i]) answer[i] += missScore.get(n) != null ? missScore.get(n) : 0;
        }
        return answer;
    }
}